.github/workflows/ci.yml

```diff
@@ -15,37 +15,74 @@ jobs:
       matrix:
         include:
 
-          - name: Python 3.7 / Django 3.2
+          # Django 2.2
+          - name: Python 3.7 / Django 2.2
             python_version: "3.7"
-            tox_env: py37-django32
+            tox_env: py37-django22
 
-          - name: Python 3.8 / Django 3.2
+          - name: Python 3.8 / Django 2.2
             python_version: "3.8"
-            tox_env: py38-django32
+            tox_env: py38-django22
 
           - name: Python 3.9 / Django 2.2
             python_version: "3.9"
             tox_env: py39-django22
 
+          # Django 3.0
+          - name: Python 3.7 / Django 3.0
+            python_version: "3.7"
+            tox_env: py37-django30
+
+          - name: Python 3.8 / Django 3.0
+            python_version: "3.8"
+            tox_env: py38-django30
+
           - name: Python 3.9 / Django 3.0
             python_version: "3.9"
             tox_env: py39-django30
 
+          # Django 3.1
+          - name: Python 3.7 / Django 3.1
+            python_version: "3.7"
+            tox_env: py37-django31
+
+          - name: Python 3.8 / Django 3.1
+            python_version: "3.8"
+            tox_env: py38-django31
+
           - name: Python 3.9 / Django 3.1
             python_version: "3.9"
             tox_env: py39-django31
 
+          # Django 3.2
+          - name: Python 3.7 / Django 3.2
+            python_version: "3.7"
+            tox_env: py37-django32
+
+          - name: Python 3.8 / Django 3.2
+            python_version: "3.8"
+            tox_env: py38-django32
+
           - name: Python 3.9 / Django 3.2
             python_version: "3.9"
             tox_env: py39-django32
 
+          - name: Python 3.10 / Django 3.2
+            python_version: "3.10"
+            tox_env: py310-django32
+
+          # Django 4.0
+          - name: Python 3.8 / Django 4.0
+            python_version: "3.8"
+            tox_env: py38-django40
+
```

Comment:
Nooooo :(
I must have been too vague in my comment. I think testing every combination of Python & Django is overkill. I think we should test 1 version of Django (the most compatible one, 3.2) on all versions of Python, and 1 version of Python (the most compatible one, 3.9) on all versions of Django. An alternative is to test old versions with old version and new versions with new versions (so e.g. py3.7+dj2.2, py3.8+dj3.0, py3.9+dj3.1, py3.9+dj3.2, py3.10+dj4.0).

I don't like quadratic things :D

Reply:
@ewjoachim it wasn't a reaction to your comment, but an initiative on my part. I'll revert to a more simple approach.